cypress-io/cypress
# chore: hide reporter when protocol is enabled

### Additional details

* Added a runtime server config item (`hideCommandLog`) that is enabled when the user has explicitly set `CYPRESS_NO_COMMAND_LOG=1` or if protocol has been enabled.
* When taking a runner screenshot (also applies to failure screenshots) and the reporter is hidden, a screenshot of the AUT in the current viewport is now taken.
  * All other existing `runner` specific logic was left as-is (i.e. `blackout` option is still ignored)

### Steps to test

* Run a test that takes a runner screenshot (`cy.screenshot({ capture: 'runner' })`) with protocol enabled.
  * Verify the reporter is not displayed while the tests are running (use `--headed` mode)
  * Verify the runner screenshot does not contain the reporter and only contains the AUT.
* Run a test that fails with `screenshotOnRunFailure` and protocol enabled.
  * Verify the failure screenshot does not contain the reporter and only contains the AUT.

### How has the user experience changed?

The Reporter UI is now hidden when protocol is enabled.
<img width="1816" alt="Screenshot 2023-08-07 at 3 02 44 PM" src="https://github.com/cypress-io/cypress/assets/2002044/0df06c47-cdad-4e88-9292-0679b84e1260">

```js
it('test 1', () => {
  cy.visit('https://example.cypress.io/commands/actions')
  cy.screenshot({ capture: 'runner', blackout: ['.action-email'] })
})
```

before:
![screenshots -- test 1](https://github.com/cypress-io/cypress/assets/2002044/c73d6ce1-9bbf-4ad7-9440-c94ad7badd1a)

after:
![screenshots -- test 1](https://github.com/cypress-io/cypress/assets/2002044/2f4f9be4-aad6-4771-94c5-4ec5bc8725af)

### PR Tasks
